openHarmony/classes/oNode.js

```diff
@@ -108,8 +121,200 @@ function oNode ( path, oSceneObject ){
+    if (ctor === oNode) return;
+    
+    // Get or create the per-type class (this does the scan if needed)
+    var TypeClass = this.getClassForType(type);
+    if (TypeClass === oNode) return; // Scan failed or not possible
+    
+    // Copy shorthand getters from the per-type class prototype to this constructor's prototype
+    // Use getOwnPropertyNames to include non-enumerable properties (shorthand getters are non-enumerable)
+    var typeProto = TypeClass.prototype;
+    var ctorProto = ctor.prototype;
+    
+    var props = Object.getOwnPropertyNames(typeProto);
+    for (var i = 0; i < props.length; i++) {
+      var prop = props[i];
+      if (ctorProto.hasOwnProperty(prop)) continue; // Don't overwrite existing
+      
+      var desc = Object.getOwnPropertyDescriptor(typeProto, prop);
+      if (desc && (desc.get || desc.set)) {
+        // It's a getter/setter, copy it
+        Object.defineProperty(ctorProto, prop, desc);
+      }
+    }
+  },
+  
+  getClassForType: function(type){
+    if (!type) return oNode;
+    if (this._classes[type]) return this._classes[type];
+
+    var tempGroupPath = null;
+    var tempNodePath = null;
+    try{
+      // Create temp group/node to discover attributes for this type
+      // Use unique name with timestamp to avoid collisions
+      var scanGroupName = '_OH_TYPE_SCAN_' + (new Date()).getTime();
+      tempGroupPath = node.add('Top', scanGroupName, 'GROUP', 0, 0, 0);
+      if (!tempGroupPath) {
+        // Failed to create temp group, fall back to base class
+        return oNode;
+      }
+      tempNodePath = node.add(tempGroupPath, 'temp', type, 0, 0, 0);
+      if (!tempNodePath) {
+        // Failed to create temp node, clean up and fall back
+        try { node.deleteNode(tempGroupPath); } catch(e) {}
+        return oNode;
+      }
+
+      var attrList = node.getAttrList(tempNodePath, 1);
+      // If attrList is null/undefined/empty, we'll use fallback keywords below
+      var baseKeywords = {};
+      var attrLength = (attrList && attrList.length) ? attrList.length : 0;
+      for (var j = 0; j < attrLength; j++) {
+        try{
+          var attr = attrList[j];
+          if (!attr || typeof attr.keyword !== 'function') continue;
+          var kw = attr.keyword().toLowerCase();
+          // Only keep top-level part (before dot) to define shorthand entry point
+          var base = kw.split('.')[0];
+          if (base === '3dpath') base = 'path3d';
+          if (base) baseKeywords[base] = true;
+        }catch(e){
+          // Skip attributes that fail
+        }
+      }
+      
+      // If no keywords found from scan, add common ones as fallback
+      var keywordCount = 0;
+      for (var k in baseKeywords) { keywordCount++; }
+      if (keywordCount === 0) {
+        // Common node attributes that most node types have
+        var commonKeywords = ['position', 'scale', 'rotation', 'offset', 'skew', 'pivot'];
```

Comment:
not sure about this. Not all nodes have these, I don't think we should offer a fallback here, throwing an error might be better, some nodes might even not have attributes which is expected (for ex multiport in and out)